build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
+ 
+ #if BUILDFLAG(ENABLE_EXTENSIONS)
+   ExtensionWebUI::RegisterProfilePrefs(registry);
+diff --git a/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc b/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
+--- a/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
++++ b/chrome/browser/profiles/chrome_browser_main_extra_parts_profiles.cc
+@@ -198,6 +198,8 @@
+ #include "chrome/browser/lacros/cert_db_initializer_factory.h"
+ #endif
+ 
++#include "components/user_scripts/browser/userscripts_browser_client.h"
++
+ namespace chrome {
+ 
+ void AddProfilesExtraParts(ChromeBrowserMainParts* main_parts) {
+@@ -452,6 +454,7 @@ void ChromeBrowserMainExtraPartsProfiles::
+   CartServiceFactory::GetInstance();
+   ProfileProtoDBFactory<cart_db::ChromeCartContentProto>::GetInstance();
+ #endif
++  user_scripts::UserScriptsBrowserClient::GetInstance();
+ }
+ 
+ void ChromeBrowserMainExtraPartsProfiles::PreProfileInit() {
+diff --git a/chrome/browser/profiles/profile_manager.cc b/chrome/browser/profiles/profile_manager.cc
+--- a/chrome/browser/profiles/profile_manager.cc
++++ b/chrome/browser/profiles/profile_manager.cc
+@@ -111,6 +111,8 @@
+ #include "extensions/common/manifest.h"
+ #endif
+ 
++#include "components/user_scripts/browser/userscripts_browser_client.h"
++
+ #if BUILDFLAG(ENABLE_SESSION_SERVICE)
+ #include "chrome/browser/sessions/session_service_factory.h"
+ #endif
+@@ -1434,6 +1436,13 @@ void ProfileManager::DoFinalInitForServices(Profile* profile,
+   }
+ 
+ #endif
++
++  user_scripts::UserScriptsBrowserClient* userscript_client =
++    user_scripts::UserScriptsBrowserClient::GetInstance();
++  if(userscript_client) {
++    userscript_client->SetProfile(profile);
++  }
++
+ #if BUILDFLAG(ENABLE_SUPERVISED_USERS)
+   // Initialization needs to happen after extension system initialization (for
+   // extension::ManagementPolicy) and InitProfileUserPrefs (for setting the
+diff --git a/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc b/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
+--- a/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
++++ b/chrome/browser/ui/webui/chrome_web_ui_controller_factory.cc
+@@ -95,6 +95,7 @@
+ #include "components/security_interstitials/content/urls.h"
+ #include "components/signin/public/base/signin_buildflags.h"
+ #include "components/site_engagement/content/site_engagement_service.h"
++#include "components/user_scripts/browser/ui/user_scripts_ui.h"
+ #include "content/public/browser/web_contents.h"
+ #include "content/public/browser/web_ui.h"
+ #include "content/public/common/content_client.h"
+@@ -578,6 +579,8 @@ WebUIFactoryFunction GetWebUIFactoryFunction(WebUI* web_ui,
+     return &NewWebUI<UserActionsUI>;
+   if (url.host_piece() == chrome::kChromeUIVersionHost)
+     return &NewWebUI<VersionUI>;
++  if (url.host_piece() == user_scripts::kChromeUIUserScriptsHost)
```

Comment:
Can WWW pages read any of these with some XHR request? No, right?

Reply:
no, I don't think it's possible, also because it's the same logic used in chrome://local-state/